fineract-core/src/main/java/org/apache/fineract/infrastructure/core/serialization/JsonParserHelper.java

```diff
@@ -518,17 +524,25 @@ public static LocalDate convertFrom(final String dateAsString, final String para
         return convertDateTimeFrom(dateAsString, parameterName, dateFormat, clientApplicationLocale).toLocalDate();
     }
 
-    public static LocalDateTime convertDateTimeFrom(final String dateTimeAsString, final String parameterName, final String dateTimeFormat,
+    public static LocalDate convertFrom(final String dateAsString, final String parameterName, final DateFormat dateFormat,
+            final Locale clientApplicationLocale) {
+
+        return convertDateTimeFrom(dateAsString, parameterName, dateFormat.getDateFormat(), clientApplicationLocale).toLocalDate();
+    }
+
+    private static LocalDateTime convertDateTimeFrom(final String dateTimeAsString, final String parameterName, String dateTimeFormat,
             final Locale clientApplicationLocale) {
 
         validateDateFormatAndLocale(parameterName, dateTimeFormat, clientApplicationLocale);
         LocalDateTime eventLocalDateTime = null;
         if (StringUtils.isNotBlank(dateTimeAsString)) {
             try {
+                dateTimeFormat = dateTimeFormat.replace("y", "u");
```

Comment:
Same as above.

Reply:
it's still necessary since I haven't changed the type everywhere. The rest will be coming in a furhter PR.